src/D2L.CodeStyle.Analyzers/Immutability/StatelessFuncAnalyzer.cs

```diff
@@ -0,0 +1,189 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+using System.Collections.Immutable;
+
+namespace D2L.CodeStyle.Analyzers.Immutability {
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	public sealed class StatelessFuncAnalyzer : DiagnosticAnalyzer {
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+			Diagnostics.StatelessFuncIsnt
+		);
+
+		public override void Initialize( AnalysisContext context ) {
+			context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction( RegisterAnalysis );
+		}
+
+		private void RegisterAnalysis( CompilationStartAnalysisContext context ) {
```

Comment:
nit: maybe later make these `static`

It's not important. I tend to look for it when there is `EnableConcurrentExecution` because then I know I only need to worry about other `static` members and captures (ahem) but like you don't have any here so it's not actually important.